vsintegration/src/FSharp.Editor/IndentationService.fs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Threading
+open System.Threading.Tasks
+open System.Linq
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Classification
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Editor.Implementation.BraceMatching
+open Microsoft.CodeAnalysis.Editor.Shared.Utilities
+open Microsoft.CodeAnalysis.Formatting
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Text
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Tagging
+
+open Microsoft.FSharp.Compiler.Parser
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+[<Shared>]
+[<ExportLanguageService(typeof<IIndentationService>, FSharpCommonConstants.FSharpLanguageName)>]
```

Comment:
Note: for some services it is preferred if you can implement the ISynchronousXXX version of it.  We have ISynchronousIndentationService for example.  This helps ensure perf for blocking features so that they are not affected by threadpool saturation.

The two services we have for this are ISynchronousIndentatoinService and ISynchronousOutliningService.

Not for indentation, you only need to implement the synchronous version.  For outlining, it's good to implement both.  The sync version is only when vs cals in synchronously to collapse things.  The async version is used for normal background updating of hte outlining regions.